packages/Search/src/details/DTK_DetailsContainers.hpp

```diff
@@ -0,0 +1,106 @@
+/****************************************************************************
+ * Copyright (c) 2012-2018 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+#ifndef DTK_DETAILS_CONTAINERS_HPP
+#define DTK_DETAILS_CONTAINERS_HPP
+
+#include <Kokkos_Macros.hpp>
+
+#include <cassert> // assert
+#include <cstddef> // size_t, ptrdiff_t
+#include <utility> // move, forward
+
+namespace DataTransferKit
+{
+namespace Details
+{
+
+// dynamic vector with fixed maximum size
+template <typename T, std::size_t N>
+class Vector
```

Comment:
This sounds like a `static_vector` in [boost](https://www.boost.org/doc/libs/master/doc/html/container/non_standard_containers.html#container.non_standard_containers.static_vector). Maybe that would be a better name than `Vector`.